From b01c615ef0da089df60b312ebf30d68ee16680b4 Mon Sep 17 00:00:00 2001
From: Andy Duan <fugang.duan@nxp.com>
Date: Wed, 7 Nov 2018 13:20:06 +0800
Subject: [PATCH 5040/5242] MLK-20258 arm64: dts: imx8qm/qxp mek: configure
 pin as RTC 32KHz clock output

commit  2da21792b59f7c7e5c95dfdaae6ba24763348737 from
https://source.codeaurora.org/external/imx/linux-imx.git

Configure imx8qm/qxp pin as RTC 32KHz clock output as the
low power clock source for some WIFI chip.

Reviewed-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi |    8 +++++++-
 arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts |    7 ++++++-
 2 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
index 60aa2f3..fe8db1a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dtsi
@@ -620,6 +620,12 @@
 				SC_P_MIPI_CSI1_GPIO0_01_LSIO_GPIO1_IO31         0x00000021
 			>;
                 };
+
+		pinctrl_wifi: wifigrp{
+			fsl,pins = <
+				SC_P_SCU_GPIO0_07_SCU_DSC_RTC_CLOCK_OUTPUT_32K	0x20
+			>;
+		};
 	};
 };
 
@@ -1082,7 +1088,7 @@
 &pciea{
 	ext_osc = <1>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pciea>;
+	pinctrl-0 = <&pinctrl_pciea &pinctrl_wifi>;
 	disable-gpio = <&gpio1 13 GPIO_ACTIVE_LOW>;
 	reset-gpio = <&gpio4 29 GPIO_ACTIVE_LOW>;
 	clkreq-gpio = <&gpio4 27 GPIO_ACTIVE_LOW>;
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
index a61e3d3..65bd8ac 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek.dts
@@ -618,6 +618,11 @@
 			>;
 		};
 
+		pinctrl_wifi: wifigrp{
+			fsl,pins = <
+				SC_P_SCU_BOOT_MODE3_SCU_DSC_RTC_CLOCK_OUTPUT_32K	0x20
+			>;
+		};
 	};
 };
 
@@ -937,7 +942,7 @@
 &pcieb{
 	ext_osc = <1>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pcieb>;
+	pinctrl-0 = <&pinctrl_pcieb &pinctrl_wifi>;
 	clkreq-gpio = <&gpio4 1 GPIO_ACTIVE_LOW>;
 	disable-gpio = <&pca9557_a 2 GPIO_ACTIVE_LOW>;
 	reset-gpio = <&gpio4 0 GPIO_ACTIVE_LOW>;
-- 
1.7.9.5

